bolt-sidecar/src/state/pricing.rs

```diff
@@ -0,0 +1,271 @@
+/// Handles pricing calculations for preconfirmations
+#[derive(Debug)]
+pub struct PreconfPricing {
+    block_gas_limit: u64,
+    base_multiplier: f64,
+    gas_scalar: f64,
+}
+
+/// Errors that can occur during pricing calculations
+#[derive(Debug, thiserror::Error)]
+pub enum PricingError {
+    /// Preconfirmed gas exceeds the block limit
+    #[error("Preconfirmed gas {0} exceeds block limit {1}")]
+    ExceedsBlockLimit(u64, u64),
+    /// Insufficient remaining gas for the incoming transaction
+    #[error("Insufficient remaining gas: requested {requested}, available {available}")]
+    /// Insufficient remaining gas for the incoming transaction
+    InsufficientGas {
+        /// Gas requested by the incoming transaction
+        requested: u64,
+        /// Gas available in the block
+        available: u64,
+    },
+    /// Incoming gas is zero
+    #[error("Invalid gas limit: Incoming gas ({incoming_gas}) is zero")]
+    InvalidGasLimit {
+        /// Gas required by the incoming transaction
+        incoming_gas: u64,
+    },
+}
+
+impl Default for PreconfPricing {
+    fn default() -> Self {
+        Self::new(30_000_000)
+    }
+}
+
+impl PreconfPricing {
+    /// Initializes a new PreconfPricing with default parameters.
+    pub fn new(block_gas_limit: u64) -> Self {
+        Self { block_gas_limit, base_multiplier: 0.019, gas_scalar: 1.02e-6 }
```

Comment:
Same here, can you add them in constants with a reference comment on top of each one?